server/channels/store/searchlayer/layer.go

```diff
@@ -111,6 +111,28 @@ func (s *SearchStore) indexUser(rctx request.CTX, user *model.User) {
 	}
 }
 
+func (s *SearchStore) indexChannelsForTeam(rctx request.CTX, teamMember *model.TeamMember) {
+	channels, err := s.Channel().GetPublicChannelsForTeam(teamMember.TeamId, 0, 100)
```

Comment:
We need to paginate and get all channels for the team. Otherwise this will work for only 100 channels.